pkg/odo/cli/deploy/deploy.go

```diff
@@ -33,7 +36,8 @@ type DeployOptions struct {
 	clientset *clientset.Clientset
 
 	// working directory
-	contextDir string
+	contextDir     string
+	commandContext context.Context
```

Comment:
I think alternatively, instead of defining a `command.Context`, you could modify the `Run` method signature to accept cmdline.Cmdline, since all the `Run` methods will require to send telemetry data.